### DISCUSSION TITLE
Using ESM causes the err_require_esm error

### DISCUSSION THREAD

**Discussion by inwardmovement** [vercel/next.js#59455] [Help]
### Summary

It seems ESM are supported ([since Next 12](https://nextjs.org/blog/next-12#es-modules-support-and-url-imports)), but with the starter template, after converting it to using ESM (see [diff](https://github.com/inwardmovement/nextjs-boilerplate/commit/ada3eda223b3056d332988106d4ba5f02786d9cf)), the build fails:

```
An error occured in `next/font`.
Error [ERR_REQUIRE_ESM]: require() of ES Module /workspaces/nextjs-boilerplate\postcss.config.js from ...\node_modules\next\dist\lib\find-config.js not supported.
```

It seems the suggestion involves modifying the code of next:
`Instead change the require of postcss.config.js in /workspaces/nextjs-boilerplate\node_modules\next\dist\lib\find-config.js to a dynamic import() which is available in all CommonJS modules.`

Any help please?
Does [this answer](https://github.com/vercel/next.js/discussions/32901#discussioncomment-1885575) mean we can't convert our next app to use ESM globally?

### Additional information

```js
⨯ app\layout.tsx
An error occured in `next/font`.

Error [ERR_REQUIRE_ESM]: require() of ES Module /workspaces/nextjs-boilerplate/postcss.config.js from /workspaces/nextjs-boilerplate/node_modules/next/dist/lib/find-config.js not supported.
Instead change the require of postcss.config.js in /workspaces/nextjs-boilerplate/node_modules/next/dist/lib/find-config.js to a dynamic import() which is available in all CommonJS modules.
    at mod.require (/workspaces/nextjs-boilerplate/node_modules/next/dist/server/require-hook.js:65:28)
    at findConfig (/workspaces/nextjs-boilerplate/node_modules/next/dist/lib/find-config.js:58:20)
    at async getPostCssPlugins (/workspaces/nextjs-boilerplate/node_modules/next/dist/build/webpack/config/blocks/css/plugins.js:89:18)
    at async /workspaces/nextjs-boilerplate/node_modules/next/dist/build/webpack/config/blocks/css/index.js:124:36
    at async /workspaces/nextjs-boilerplate/node_modules/next/dist/build/webpack/loaders/next-font-loader/index.js:86:33
    at async Span.traceAsyncFn (/workspaces/nextjs-boilerplate/node_modules/next/dist/trace/trace.js:147:20)
```

### Example

https://github.com/inwardmovement/nextjs-boilerplate

**Comment by muhfiz** [comment]
Have you found the solution?

**Reply by r34son** [reply]
https://github.com/vercel/next.js/issues/34448#issuecomment-2002513315

**Comment by hb20007** [comment]
I created an Issue for this: https://github.com/vercel/next.js/issues/63995 and was told in the comments that it's now **fixed** in the latest canary.
